File: projects/07/array_processing.rb
# arrayProcessing.java is a class that provides two methods: printArray and readArray
# 
# Completed by: Josiah Ryan
# Date        : 03/22/21
##########################################################

# readArray prompts user for the values to store into the array
# Receive: 	an array 
# Does   :  Stores user inputed values into the array
######################################################

def readArray (anArray)
  print "Enter array elements:\n"
  for i in 1..anArray.length() do             #loops through the array and stores user inputed values into array
    print "  >"
    item = gets()
    anArray[i] = item
  end 
end 

# printArray prints the array passed to it
# Receive:  an array 
# Does   :  prints the array
################################################
def printArray (anArray)
  print "your array is: \n"
  for i in 1..anArray.length() do           #loops through array and prints values from the array
    puts anArray[i]
  end
end

def main
  print "Enter the size of the array ";
  size = gets.to_i                        #gets size
  anArray = Array.new(size)  
  readArray(anArray)                      #reads in array
  printArray(anArray)                     #prints off array
end

main

File: projects/07/arrayProcessing.java
/* arrayProcessing.java is a class that provides two methods: printArray and readArray
 * 
 * Completed by: Josiah Ryan
 * Date        : 03/22/21
 ***************************************************************/

import java.util.Scanner;
import java.lang.Math;

public class arrayProcessing
{
  // main program
  public static void main(String[] args)
    {
      Scanner keyboard = new Scanner(System.in);
      // prompt for the string
      System.out.println("\nEnter array size:");
      String str = keyboard.nextLine();
      int size = Integer.parseInt(str);
      double[] array = new double[size];     //creates an array of user specified size
      readArray(array);
      printArray(array);
      }
  
  /* readArray prompts user for the values to store into the array
   * Receive: 	an array 
   * Does   :  Stores user inputed values into the array
   *****************************************************/

  public static void readArray(double[] array) {	
    String moveStr;
    Scanner keyboard = new Scanner(System.in);
    System.out.println("Enter values into the array:");
      for (int i = 0; i<array.length; i++) {            //loops through the array reading in values from the user
       System.out.print("  >");
       moveStr = keyboard.nextLine();
       array[i] = Integer.parseInt(moveStr);
      }
  }
  
   /* printArray prints the array passed to it
   * Receive:  an array 
   * Does   :  prints the array
   *****************************************************/

  public static void printArray(double[] array) {	
    System.out.println("\nYour Array is:");

    for (int i = 0; i<array.length; i++) {
        System.out.println(array[i]);
       }
  }

}
